UnityProject/Assets/Scripts/Core/UniversalObjectPhysics.cs

```diff
@@ -0,0 +1,1610 @@
+using System;
+using System.Collections.Generic;
+using Core.Editor.Attributes;
+using HealthV2;
+using Items;
+using Messages.Server.SoundMessages;
+using Mirror;
+using Objects;
+using Objects.Construction;
+using UnityEngine;
+using UnityEngine.Events;
+using Util;
+using Random = UnityEngine.Random;
+
+public class UniversalObjectPhysics : NetworkBehaviour, IRightClickable
+{
+	//TODO parentContainer Need to test
+	//TODO Maybe work on conveyor belts and players a bit more
+	//TODO Sometime Combine buckling and object storage
+	//=================================== Maybe
+	//TODO move IsCuffed to PlayerOnlySyncValues maybe?
+	//=================================== balance Maybe
+	//=============================================== TODO some time
+	//TODO Make space Movement perfect ( Is pretty much good now )
+	//TODO after thrown not synchronised Properly need to synchronise rotation
+	//TODO When throwing rotation Direction needs to be set by server
+	//=============================================== Definitely
+
+	public const float DEFAULT_PUSH_SPEED = 6;
+
+	/// <summary>
+	/// Maximum speed player can reach by throwing stuff in space
+	/// </summary>
+	public const float MAX_SPEED = 25;
+
+	public const int HIGH_SPEED_COLLISION_THRESHOLD = 13;
+
+
+	//public const float DEFAULT_Friction = 9999999999999f;
+	public const float DEFAULT_Friction = 15f;
+	public const float DEFAULT_SLIDE_FRICTION = 9f;
```

Comment:
Should these be serialised so different objects have different friction or are these some constants needed for some math